src/test/java/httpServer/RouterTest.java

```diff
@@ -15,34 +15,41 @@
-        ResponseBuilder responseBuilder = new ResponseBuilder();
+        String expectedResponse = "HTTP/1.1 " + StatusCodes.OK.httpResponse + Constants.CRLF + "Allow: GET" + Constants.CRLF + "Content-Length: 0" + Constants.CRLF;
+        HTTPResponseWriter responseBuilder = new HTTPResponseWriter();
         Router router = new Router(HttpServer.getRoutes(), responseBuilder);
-        String returnedCode = router.getResponse(request);
-        System.out.print(returnedCode);
-        assertEquals(expectedStatusCode, returnedCode);
+        String returnedResponse = router.getResponse(request);
+        assertEquals(expectedResponse, returnedResponse);
     }
 
     @Test
     void testRouterReturns404ForInvalidRoute() throws IOException {
         HTTPRequest request = new HTTPRequest("GET", "/anything_invalid", "HTTP/1.1");
-        String expectedStatusCodes = StatusCodes.PAGE_NOT_FOUND.httpResponse + Constants.CRLF + "Content-Length: 0" + Constants.CRLF;
-        ResponseBuilder responseBuilder = new ResponseBuilder();
+        String expectedResponse = "HTTP/1.1 " + StatusCodes.PAGE_NOT_FOUND.httpResponse + Constants.CRLF + "Content-Length: 0" + Constants.CRLF;
+        HTTPResponseWriter responseBuilder = new HTTPResponseWriter();
         Router router = new Router(HttpServer.getRoutes(), responseBuilder);
-        String returnedCode = router.getResponse(request);
-        System.out.print(returnedCode);
-        assertEquals(expectedStatusCodes.toString(), returnedCode);
+        String returnedResponse = router.getResponse(request);
+        assertEquals(expectedResponse.toString(), returnedResponse);
     }
 
     @Test
-    void testRouterReturns405ForInvalidMethod() throws IOException {
-        HTTPRequest request = new HTTPRequest("GET", "/head_request", "HTTP/1.1");
-        String expectedStatusCodes = StatusCodes.NOT_ACCEPTABLE.httpResponse + Constants.CRLF + "Allow: HEAD, OPTIONS" + Constants.CRLF +  "Content-Length: 0" + Constants.CRLF;;
-        ResponseBuilder responseBuilder = new ResponseBuilder();
+    void testRouterReturns200ForValidOptionsRequest() throws IOException {
+        HTTPRequest request = new HTTPRequest("OPTIONS", "/head_request", "HTTP/1.1");
+        String expectedResponse = "HTTP/1.1 " + StatusCodes.OK.httpResponse + Constants.CRLF + "Allow: HEAD, OPTIONS" + Constants.CRLF +  "Content-Length: 0" + Constants.CRLF;;
+        HTTPResponseWriter responseBuilder = new HTTPResponseWriter();
         Router router = new Router(HttpServer.getRoutes(), responseBuilder);
-        String returnedCode = router.getResponse(request);
-        System.out.print(returnedCode);
-        assertEquals(expectedStatusCodes, returnedCode);
+        String returnedResponse = router.getResponse(request);
+        assertEquals(expectedResponse, returnedResponse);
+    }
+
+    @Test
+    void testRouterReturns200ForValidHeadRequest() throws IOException {
```

Comment:
This is very literal and more about the specifics of your router than test cases that check the logic of your code.

You want to return 200 for a valid request to a resource with one allowed method (your first case). And also to return 200 for a valid request to a resource with _multiple_ allowed methods (that's the case you were erroring out on).

Reply:
Pretty sure this test case is accurately covered now with tests for `HEAD` and `OPTIONS` to `/head_request` as well as the `405` for `GET` to `/head_request`, but let me know if I misunderstood! The error case of only checking the first list case should be taken care of.

Reply:
Sorry, my comment wasn't clear! I meant the naming of the tests was too literal and not very helpful about what kind of cases you were trying to test.